### PR TITLE
render empty oauth scopes

### PR DESCRIPTION
**Pull request by khart-twilio**
see https://github.com/pb33f/libopenapi/issues/289

This is my naive attempt to fix the issue.  It works but it may have negative side-effects I'm not aware of.

### CODE REVIEW

File: orderedmap/builder.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ type findValueUntyped interface {
 // ToYamlNode converts the ordered map to a yaml node ready for marshalling.
 func (o *Map[K, V]) ToYamlNode(n NodeBuilder, l any) *yaml.Node {
 	p := utils.CreateEmptyMapNode()
+	if o != nil {
```
Comment (daveshanley) [inline orderedmap/builder.go:39]: Can you add a small test for this, as this new block will reduced coverage, if you pull the project again and update the workflow config (upgraded the codecov action) You should see the coverage on this PR.

**Comment by khart-twilio** [comment]
@daveshanley One edge case I noticed when testing:

I had: 
```
...
      requestBody:
        content:
          application/json:
            schema:
              type: array
              items:
                type: object
                properties:
                  value:
                    type: null # null isn't allowed here I don't think according to the spec
```
And with these changes in libopenapi it is now being processed and spit out as

```
...
                properties:
                  value:
                    type: "ObjectNode"
```

It doesn't matter for this case since using `null` was invalid and I changed it to something else, but I'm wondering if it might matter in other places.  I'm not familiar with how things like `null`, `true`, `false` come through as yaml.Node object when deserialized and if I'd need to handle these cases in the changes.

File: orderedmap/builder.go (hunk 1 above)
Comment (khart-twilio) [reply orderedmap/builder.go:39]: I merged the latest change from main branch and ran `go test -v -coverprofile cover.out ./...` locally and see 100% coverage.  In the status on this PR I see `1 workflow awaiting approval`, which I assume means you need to approve something before the coverage check can run here, but I'm unfamiliar.  I'd appreciate if you clarify for me.

Comment (khart-twilio) [reply orderedmap/builder.go:39]: whoops, nevermind I was reading the test output incorrectly.  I'll try to figure it out.

**Comment by daveshanley** [comment]
> @daveshanley One edge case I noticed when testing:
> 
> I had:
> 
> ```
> ...
>       requestBody:
>         content:
>           application/json:
>             schema:
>               type: array
>               items:
>                 type: object
>                 properties:
>                   value:
>                     type: null # null isn't allowed here I don't think according to the spec
> ```
> 
> And with these changes in libopenapi it is now being processed and spit out as
> 
> ```
> ...
>                 properties:
>                   value:
>                     type: "ObjectNode"
> ```
> 
> It doesn't matter for this case since using `null` was invalid and I changed it to something else, but I'm wondering if it might matter in other places. I'm not familiar with how things like `null`, `true`, `false` come through as yaml.Node object when deserialized and if I'd need to handle these cases in the changes.

I have no idea where `ObjectNode` is coming from. It's not part of `libopenapi` 

Because I have no idea, let's just proceed, if someone else finds this issue then we can investigate further.
